index.md

```diff
@@ -2,9 +2,286 @@
+
+In operations a similar pattern will be used in that the raw data and formal data products must be stored in read-only locations, but with the additional constraint that users will have [quotas](#quotas) on the amount of storage they can use.
+
+There are many options where a dataset could be stored when the user issues `butler.put()`:
+
+1. Their personal POSIX storage space.
+2. Per-user and per-group buckets in the cloud or at the USDF.
+3. A general user bucket in the cloud or at the USDF per butler where `u/$USER/` is enforced.
+
+Option #1 allows a single storage area for all the user's data products and simplifies the calculation of quotas.
+It does not provide any means of supporting collaborative groups and that likely means this option is not tenable other than in the narrow "workspace" concept described in {cite:p}`DMTN-271`.
+
+The remaining options will all involve the server generating signed URLs for writes.
+
+Option #2 solves the collaboration problem at the expense of creating tens of thousands of buckets.
+Option #3 is similar to how we currently handle user data products but does require more work to handle quotas than option #2.
+
+The choice between USDF and Cloud storage buckets depends on the cost of cloud storage versus USDF storage (based on the expected size of user data products) and how user batch {cite:p}`DMTN-223` interacts with cloud users.
+
+All user batch processing will take place at the USDF such that any user products to be used as inputs for that processing must be available at the USDF (the generated graph should not use signed URLs to locate the files in the cloud since signing happens in the server and the server is not available at batch execution time and signed URLs can not be stored in the graph because of expiry times) and all the data products created by the processing must be made available to the user and stored in their user or group collection.
+This constraint from batch processing suggests that USDF may be the best option since the files can be read from and written to the final location at USDF without requiring transfers.
+
+```{warning}
+It has been noted that Weka can only support up to 10,000 buckets.
+This is similar to the expected number of users and implies that if Weka is required as the USDF object store interface and user products are to be stored at USDF, then option #2 can not be used.
+```
+
+One additional thing to consider is how to handle user storage with multiple data releases.
+The first two options listed would need to have additional information in the file hierarchy to reference the relevant parent butler repository.
+The final option would use a server configuration to indicate where the specific user bucket is located as is done for DP0.2.
+
+#### Quotas
+
+Dataset file quotas for RSP users' POSIX file space will be monitored by the RSP infrastructure itself.
```

Comment:
This is probably out of scope, but I think it'd be fantastic if we could let users trade a smaller amount of POSIX storage for a cost-equivalent amount of Cloud object storage.